phalcon/di/service/serviceregistry.zep

```diff
@@ -0,0 +1,115 @@
+namespace Phalcon\Di\Service;
+
+use Phalcon\DiInterface;
+use Phalcon\Di\Exception;
+use Phalcon\Di\ServiceInterface;
+use Phalcon\Config\Adapter\Yaml;
+
+/**
+ * Phalcon\Di\Service\ServiceRegistry
+ *
+ **/
+class ServiceRegistry
+{
+
+	/**
+	 *
+	 */
+	protected di;
+
+	/**
+	 *
+	 */
+	public function __construct(<DiInterface> di)
+	{
+		let this->di = di;
+	}
+
+	/**
+	 *
+	 */
+	public function registerDir(dirPath)
+	{
+		var dirPathItem;
+
+		switch (gettype(dirPath)) {
+			case "string":
+				this->_registerDir(dirPath);
+				break;
+			case "array":
+			case "object":
+				for dirPathItem in iterator(dirPath) {
+					this->_registerDir(dirPathItem);
+				}
+				break;
+			default:
+				throw new Exception("Invalid services.");
+				break;
+		}
+	}
+
+	/**
+	 *
+	 */
+	protected function _registerDir(string! dirPath)
+	{
+		var dir, fileinfo, extension, serviceNameSection, forceShared, serviceName;
+
+		let dir = new \DirectoryIterator(dirPath);
+		for fileinfo in iterator(dir) {
+			if (fileinfo->isDir()) {
```

Comment:
```zep
if unlikely fileinfo->isDir() === true {
    // ...
}
```

Also I'm pretty sure we don't need to scan this dir in each request

Reply:
How about **`.`** and **`..`** ?

Reply:
Well I don't know about caching here.  I don't want to handle any directory (current or parent) and for now I think that its best not to recurse.  It only reads files.

Reply:
In this case I propose consider such  a design with which user can easily inject a caching mechanism. For example Decorator Pattern

Reply:
Ok well this should be before the DI has been setup, since this is is intended to be a possible first step for getting services going.

I won't be working on this for 7 days and then I will take down this pull request and do more work on it and then resubmit just two clean commits.

Reply:
btw what is the time difference between scanning a directory with 1-25 files in it versus reading a single file?  Also how would changes to individual files be detected?

If someone could propose to me a mechanism for caching this data then it would give me something to work with.

Reply:
Php already caches stuff like this.